internal/imex/imex.go

```diff
@@ -0,0 +1,108 @@
+/**
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package imex
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"k8s.io/klog/v2"
+
+	spec "github.com/NVIDIA/k8s-device-plugin/api/config/v1"
+)
+
+// Channels represents a set of IMEX channels.
+type Channels []*channel
+
+// channel represents an IMEX channel.
+type channel struct {
+	id      string
+	devRoot string
+}
+
+// GetChannels returns the set of channels for the given config.
+// If the selection of the default IMEX channel is disabled no channels are returned.
+func GetChannels(config *spec.Config, devRoot string) (Channels, error) {
+	var channels Channels
+	for _, channelID := range config.Imex.ChannelIDs {
+		channel := channel{
+			id:      fmt.Sprintf("%d", channelID),
+			devRoot: devRoot,
+		}
+		if exists, err := channel.exists(); !exists {
+			if config.Imex.Required {
+				return nil, errors.Join(err, fmt.Errorf("requested IMEX channel %v does not exist", channel.Name()))
+			}
+			klog.Warningf("Ignoring requested IMEX channel %v (%v)", channel.Name(), err)
+			continue
+		}
+		klog.Infof("Selecting IMEX channel %v", channel.Name())
+		channels = append(channels, &channel)
+	}
+	return channels, nil
+}
+
+// GetPaths returns the paths for all IMEX channels in a set of IMEX channels.
+func (c *Channels) GetPaths() []string {
+	if c == nil {
+		return nil
+	}
+	var paths []string
+	for _, channel := range *c {
+		paths = append(paths, channel.Path())
+	}
+	return paths
+}
+
+// ID returns an identifier for the channel.
+func (c channel) ID() string {
+	return c.id
+}
+
+// Name returns the channel name.
+func (c channel) Name() string {
+	return "channel" + c.id
+}
+
+// Path returns the absolute path to an IMEX channel.
+func (c channel) Path() string {
+	return filepath.Join("/dev/nvidia-caps-imex-channels", c.Name())
+}
```

Comment:
As suggested above, can we just turn all of these into exported fields on the struct (and make it public) rather than have these helper functions?